apps/addons/models.py

```diff
@@ -497,7 +494,14 @@ def delete(self, msg='', reason=''):
     @classmethod
     def initialize_addon_from_upload(cls, data, is_listed=True):
         fields = cls._meta.get_all_field_names()
+        # Reclaim GUID from deleted add-on.
+        try:
+            old_guid_addon = Addon.unfiltered.get(guid=data['guid'])
+            old_guid_addon.update(guid=None)
```

Comment:
I don't understand your point: i was proposing to change it to "guid-reused-by..." straight away here.

But you raise a valid point: what about the use case where someone deletes his add-on several times? Then you'd have multiple deleted add-ons with the same guid (guid-reused-by...), which will fail. This needs a solution, and a test ;)